### PR TITLE
Request Transaction Capture

### PR DESCRIPTION
**Pull request by rafsAcorsi**
Modified passing parameter to the method captures the request, Added raise Exception if the amount has no value.

### CODE REVIEW

File: pagarme/transaction.py [hunk 1]
```diff
@@ -66,9 +66,14 @@ def handle_response(self, data):
     def capture(self):
         if self.id is None:
             raise NotBoundException('First try search your transaction')
+        if self.amount is None:
+            raise NotBoundException('First add value to variable amount')
```
Comment (mbodock) [inline pagarme/transaction.py]: You shouldn't use `NotBoundException`. How about a new Exception? Something with a better name.

File: pagarme/transaction.py [hunk 2]
```diff
@@ -66,9 +66,14 @@ def handle_response(self, data):
...
         url = self.BASE_URL + '/' + str(self.id) + '/capture'
-        data = {'api_key': self.api_key}
-        pagarme_response = requests.post(url, data=data)
+        params = {
+            'api_key': self.api_key,
+            'amount': self.amount
+        }
+        pagarme_response = requests.post(url, params=params)
```
Comment (mbodock) [inline pagarme/transaction.py]: Try don't use `params` on `post`. The `data` is used to sent information inside the body, and `params` is used to send information with querystring (usualy in `GET`).

**Comment by mbodock** [comment]
Hi @rafsAcorsi,

Since you're adding a new feature, how about some new tests to match this new code?
We can't allow our coverage decrease.

Regards.

File: pagarme/transaction.py (hunk 2 above)
Comment (rafsAcorsi) [inline pagarme/transaction.py]: I agree with you, but the api has a problem and does not identify the parameters passed in the body, only in the querystring.

**Comment by rafsAcorsi** [comment]
Tests Done.

**Comment by rafsAcorsi** [comment]
@mbodock tudo ok?

**Comment by mbodock** [comment]
@rafsAcorsi looks fine to me.
I just want to test the `params` vs `data` in the requests.

Also I need to arrange with @lerrua the next release.

Just give me a couple days.
Regards

**Comment by rafsAcorsi** [comment]
Ok @mbodock  @cassiobotaro manda um abraço para @lerrua!!kkkk

**Comment by rafsAcorsi** [comment]
According to the API of them, metadata can be passed through when capture is made transparent checkout see: [https://docs.pagar.me/checkout/#capturando-a-transacao](https://docs.pagar.me/checkout/#capturando-a-transacao)

**Comment by mbodock** [comment]
@rafsAcorsi look's fine to me, do you mind squash your commits in just one?

**Pull request closed without merging**